kong/db/dao/init.lua

```diff
@@ -458,14 +459,13 @@ local function generate_foreign_key_methods(schema)
           validate_options_type(options)
         end
 
-        local ok, errors = self.schema:validate_field(field, foreign_key)
+        local ok, err = self.schema:validate_field(field, foreign_key)
```

Comment:
Looks like you changed a lot of `errors` variables to `err` in this module: these were named `errors` on purpose, to match their naming in the `kong.db.schema.init` module, where `errors` is used to refer for the errors table with the same type produced by that module, and differentiate them from `err`, which is used for a single error (of type string), and from `err_t` which is the error table (of a different type) produced by the `kong.db.errors` module.

There are three error things of different types going on in `kong.db.dao.init` (two of which are tables with different structures). The names were kept different to make it easier to keep track which is which. (Yeah, it sucks we don't have types to do this work properly, so we try to fake it with naming conventions ;) )

I personally would prefer these naming changes reverted, but I don't consider this a blocker for merging if you prefer to keep this change (the above was more of an explanation as to why the names were like this in the first place).

Reply:
Ok, made those changes, also tried to find other places to make it consistent.